Issue #24 - insert DTO type ENUM as string

bwengert79 opened the pull request:
Issue #24 
When serializing a DTO that contains a DTO_FIELD whose type is `Enum<xxx>AsString`, the resulting string ultimately goes out of scope. This will result in a failure when trying to insert this type of value into a database.

The solution is to utilize Serializer::OutputData::databuffer. Before the enum string goes out of scope, allocate a dataBuffer and copy the value to it.

EnumAsStringTest is included as a unit test. 

If you run the unit test before applying the Serializer change you can see the failure take place.

Regarding Oatpp 1.4.0
My environment is set up using Oatpp 1.4.0.  
I updated oatpp-postresql to 1.4.0 to explore.
I then discovered issue #24 and created the solution in the 1.4.0 environment.

bwengert79 committed:
update to oatpp140

bwengert79 committed:
removed comment regarding oatpp140

bwengert79 committed:
issue #24 - insert DTO type ENUM as string

bwengert79 committed:
Merge branch 'oatpp140' into develop
Update to oatpp 1.4.0

bwengert79 committed:
run test EnumAsStringTest

bwengert79 committed:
Merge branch 'issue24' into develop
issue #24 - insert DTO type ENUM as string
Include test case EnumAsStringTest

lganzzzo commented:
Hey @bwengert79 ,
It's a great PR!

**Let's try to run CI:**
Please change `docker-compose` to `docker compose` without '-' in two places
https://github.com/oatpp/oatpp-postgresql/blob/master/azure-pipelines.yml#L16
Let's see if it builds

bwengert79 committed:
Update azure-pipelines.yml

Version two of the Docker Compose is invoked via docker compose
Removed hyphen from docker-compose

bwengert79 commented:
Got past the docker compose issue.
The build is now failing because it looks like a cmake version issue.
"CMake 3.20 or higher is required.  You are running version 3.11.1"
I changed cmake_minimum_required to 3.20 in CMakeList.txt to match what's in the oatpp repo.
Is oatpp-postgresql building in a different environment that oatpp?

bwengert79 committed:
Revert cmake min back to 3.1 to resolve CI issue

bwengert79 committed:
Revert "Revert cmake min back to 3.1 to resolve CI issue"

This reverts commit 633c2465444c1d55f5e0da3e62577b1229f4135b.

bwengert79 commented:
Changing cmake min to 3.1 got us further in the build process. However, the next stumbling block is oatpp 140 is not available in the current CI environment. Perhaps this PR needs to wait for the CI environment to match that of oatpp 140.

lganzzzo commented:
Hey @bwengert79 ,

Yes, the current implementation of CI pipeline is outdated and isn't flexible.
I'll merge this PR as is, and later will rebuild the CI pipeline.
